api/complete-multipart: fixes and tests.

hackintoshrao opened the pull request:
- Removing the logic of sending white space characters during complete multipart upload. 
- This simplifies the code and helps to meet the requirements mentioned at https://github.com/minio/minio/issues/2690 in a cleaner way.
- Tests for New Multipart Upload and Complete Multipart Upload API handlers.
- Tests for Delete Object Handler. 
- Fixes https://github.com/minio/minio/issues/2690 and https://github.com/minio/minio/issues/2671  and https://github.com/minio/minio/issues/2669 . 

With detailed discussion with @krishnasrinivas , we found out this approach to be the simplest amongst the other alternatives.

krishnasrinivas reviewed (commented):
@hackintoshrao @harshavardhana the complication for completemultipart upload is arising because we have to accomodate the sendwhitespace() function. sendwhitespace() is not useful for XL, it was written because of slow concat process of FS. But since we now do append in background as parts come, FS's completemultipart() is also quick now. That means we can just remove sendwhitespace() functionality and avoid all complications.

Downside is we will not send whitespace chars for the slow concat fall back in FS. But it's a compromise we can make to keep code simple. Because anyhow we saw cloudberry timeout even if we sent white space chars. So for well behaved client uploads we work just fine.

harshavardhana commented on cmd/api-response-multipart.go:
```diff
@@ -20,6 +20,21 @@ package cmd
 import "net/http"
 
+// Represents additional fields necessary for ErrPartTooSmall S3 error.
+type completeMultipartAPIError struct {
```
Do not introduce these custom errors just yet, i am fixing this differently. For now do not validate other fields other than the common ones that occur in `APIErrorResponse`

hackintoshrao replied:
OK, for now I'll make changes to send usual error response.

harshavardhana commented on cmd/object-handlers.go:
```diff
@@ -806,28 +837,19 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 		doneCh <- struct{}{}
 	}(doneCh)
 
 	sendWhiteSpaceChars(w, doneCh)
```
Let's remove this and we will solve all the problems and we will not need to change the function signature of `CompleteMultipartUpload`

hackintoshrao replied:
So I'll keep the CompleteMultipart API signature as it was and remove the sending of white space characters and running complete multipart in a new routine.

Commit:
api/complete-multipart: tests and simplification.

- Removing the logic of sending white space characters.
- Fix for incorrect HTTP response status for certain cases.
- Tests for New Multipart Upload and Complete Multipart Upload.

Commit:
tests: test for Delelete Object API handler
